### PR TITLE
test: rewrite cutler to be testable

### PR DESCRIPTION
**Pull request by eseidel**
I removed the 'rebase' command for now since it was dead.
'rebase' should be re-written later to just print the commands
needed since having it actually drive the rebase was a dream.
I separated "Repo" from "Checkout" and removed the global config object.
I moved to using a global logger and package scoped to match
other packages.
I added a bunch of tests.  This still doesn't have 100% coverage
but it should
be 100% testable now.

### CODE REVIEW

File: packages/cutler/lib/git_extensions.dart [hunk 1]
```diff
@@ -50,20 +50,35 @@ extension RepoCommands on Repo {
...
       'git',
-      ['fetch', '--all'],
+      ['fetch', '--all', '--tags'],
       workingDirectory: workingDirectory,
     );
   }
 
   /// Returns a [Version] for the given [hash].
-  Version versionFrom(String hash, {bool lookupTags = true}) {
+  Version versionFrom(String commitish, {bool lookupTags = true}) {
```
Comment (bryanoltman) [inline packages/cutler/lib/git_extensions.dart]: > commitish

Typo?

**Review by bryanoltman — APPROVED** [review]

File: packages/cutler/lib/git_extensions.dart (hunk 1 above)
Comment (eseidel) [reply packages/cutler/lib/git_extensions.dart]: https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefcommit-ishacommit-ishalsocommittish

Comment (bryanoltman) [reply packages/cutler/lib/git_extensions.dart]: Huh! TIL

Comment (bryanoltman) [reply packages/cutler/lib/git_extensions.dart]: Probably should update the doc comment to reflect the argument rename

Comment (eseidel) [reply packages/cutler/lib/git_extensions.dart]: Fixed.

**Comment by eseidel** [comment]
I ended up doing a massive rewrite of cutler.

**Review by felangel — APPROVED** [review]

**Pull request merged**